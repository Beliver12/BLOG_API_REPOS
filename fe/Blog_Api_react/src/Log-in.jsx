import { useState, useEffect} from 'react'

import { Link, Navigate } from "react-router-dom";


const LogIn = () => {
    debugger;
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [message, setMessage] = useState('')
    
 
    const handleSubmit  = async (event) => {
        debugger;
      
        event.preventDefault();
       const data = {
        password: password,
        username: username
       }

       const url = "http://localhost:3000/login"
       
       const options = {
        method: "POST",
        body: JSON.stringify(data),
        credentials: 'include',
        headers: {
            "Content-Type": "application/json"
        }
       }
     
       fetch(url, options)
       .then(res => res.json())
       .then((data) => {
        setUsername('')
        setPassword('')
        console.log(data);
       
        localStorage.setItem("accessToken", data.token);
        setMessage('true')
       }).catch((error) => console.error("Error", error))

       
     
      
    }
    if(message){
        return <Navigate to="/"/>
    }
    const handleLogout = () => {
        localStorage.removeItem("accessToken");
       }
    return (
        <>
        <div className='log-in'>
            <h1>Log-in</h1>

            <form onSubmit={handleSubmit} method='POST' >
                <input type="text"  name='username' value={username} onChange={(e) => setUsername(e.target.value)}/> 
                <input type="password"  name="password" id="" value={password} onChange={(e) => setPassword(e.target.value)}/>
        
        <button type='submit'>Log-in</button>
        
            </form>
            <button onClick={handleLogout}>Log-out</button>
            <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
        </ul>
        </nav>
        </div>
        </>
    )
}

export default LogIn