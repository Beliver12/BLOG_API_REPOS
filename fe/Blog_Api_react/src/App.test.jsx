import { describe, it, expect} from 'vitest';
import { createRoutesStub } from "react-router";
import {
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { LogIn } from "./Log-in";

test("LoginForm renders error messages", async () => {
  const USER_MESSAGE = "Username is required";
  const PASSWORD_MESSAGE = "Password is required";

  const Stub = createRoutesStub([
    {
      path: "/log-in",
      Component: LogIn,
      action() {
        return {
          errors: {
            username: USER_MESSAGE,
            password: PASSWORD_MESSAGE,
          },
        };
      },
    },
  ]);

  // render the app stub at "/login"
  render(<Stub initialEntries={["/log-in"]} />);

  // simulate interactions
  userEvent.click(screen.getByText("log-in"));
  
});

describe('something truthy and falsy', () => {
  it('true to be true', () => {
    expect(true).toBe(true);
  });

  it('false to be false', () => {
    expect(false).toBe(false);
  });
});