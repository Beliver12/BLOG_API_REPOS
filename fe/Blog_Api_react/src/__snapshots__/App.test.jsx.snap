// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`start page test > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar"
      >
        <nav>
          <h1>
            MY-BLOG
          </h1>
          <ul>
            <a
              data-discover="true"
              href="/"
            >
              Home
            </a>
            <a
              data-discover="true"
              href="/posts"
            >
              Posts
            </a>
            <a
              data-discover="true"
              href="/signup"
            >
              Signup Page
            </a>
            <a
              data-discover="true"
              href="/log-in"
            >
              Log-in Page
            </a>
          </ul>
        </nav>
        <div
          class="main"
        >
          <div
            class="Languages"
          >
            <h1>
              Created with:
            </h1>
            <img
              alt=""
              src="..//icons8-nodejs-144.png"
            />
            <a
              href="https://icons8.com/icons"
            >
              <strong>
                Nodejs icon by Icons8
              </strong>
            </a>
            <img
              alt=""
              src="..//icons8-express-js-128.png"
            />
            <a
              href="https://icons8.com/icons"
            >
              <strong>
                Express Js icon by Icons8
              </strong>
            </a>
            <img
              alt=""
              src="..//icons8-react-100.png"
            />
            <a
              href="https://icons8.com/icons"
            >
              <strong>
                React icon by Icons8
              </strong>
            </a>
            <img
              alt=""
              src="..//icons8-json-web-token-144.png"
            />
            <a
              href="https://icons8.com/icons"
            >
              <strong>
                JSON Web Token icon by Icons8
              </strong>
            </a>
            <a
              href="https://icons8.com/icons"
            >
              Square icon by Icons8
            </a>
          </div>
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
